### DISCUSSION TITLE
Réflexions scénario export corpus

### DISCUSSION THREAD

**Discussion by marviro** [EcrituresNumeriques/stylo#1075] [Ideas]
## Problématique

Qu'est-ce qu'un ensemble de documents Stylo?

Stylo est pensé pour des articles individuels. L'unité est l'article. Cela implique que par exemple des information sur les dossiers se trouvent dans les métadonnées de chaque article.

Mais :

1. le revues ont de plus en plus le besoin de traiter des lots d'articles
2. Stylo est utilisé aussi pour faire des livres et des thèses (dans ce cas le document n'est pas un article mais un chapitre)

## Cas d'usage

Nous avons deux cas d'usages principaux avec des sous-cas:

1. Corpus d'articles
  a. pour un dossier. Exporter un dossier de revue, par exemple en XML Métopes, pour publication en lot
  b. pour une revue entière: tous les articles d'une revue, par exemple pour produire automatiquement le site d'une revue
2. Corpus de chapitres
  a. pour faire un livre
  b. pour faire un mémoire ou une thèse

## Question métadonnées

1. Comment produire les métadonnées corpus dans le cas de corpus d'article? **Nous avons décidé de demander une action: récupère les métadonnée à partir d'un article du corpus au choix**
2.

**Comment by thom4parisot** [comment]
J'ai déplacé ça dans les Discussions dans la catégorie Idées, dans l'idée de laisser les "issues" pour la description des implémentations (donc qu'on a déjà itéré dans les idées et qu'on est plus au clair sur ce qu'on doit faire).

**Comment by RochDLY** [comment]
## Introduction : du _Book_ au _Corpus_ 

La fonctionnalité _Corpus_ de Stylo remplace l'ancienne fonctionnalité _Book_ qui, comme son nom l'indique, était utilisée pour produire des livres à partir d'un script bash.
_Book_ récupérait un ensemble d'articles regroupés au moyen d'un tag qu'on pouvait leur ajouter.
À l'intérieur de ce tag, les articles étaient ordonnés manuellement grâce au comportement par défaut de Stylo de tri par ordre alphabétique des articles dans son gestionnaire d'articles.
Ainsi les articles étaient ordonnés via un tag et une gestion de leur nom préfixé par un indicatif numérique (1, 2, 3, 4, etc..) pour préserver l'ordre souhaité lors de la transformation en livre.

Grâce à _Book_, il était possible d'exporter des ensembles d'articles Stylo en un seul document (avec toutes les fonctionnalités que ça implique de calcul de la pagination, de la génération de la table des matières, etc.).
Toutefois, cette fonctionnalité bricolée est arrivée à sa limite lorsque nous avons commencé à vouloir exporter des ensembles de documents selon divers comportements.

C'est à ce stade qu'arrive la fonctionnalité _Corpus_.

Là où _Book_ décrivait l'objet produit par la fonctionnalité, le nom de _Corpus_ est plus vague, c'est un choix volontaire (cf #781) discuté avec toute l'équipe.
Dans le ticket #781, @ggrossetie nous rappelle qu'un corpus est un « ensemble fini de textes choisi comme base d'une étude ».
Nous avons fait ce choix parce qu'en tant qu'ensemble fini de textes, un corpus peut tout à fait désigner un livre, un numéro de revue, une thèse ou un mémoire, un ensemble de billets de blog ou des notes que l'on regroupe.

En comparaison de _Book_, _Corpus_ offre plusieurs changements structurels :

1. Les _Corpus_ existe en base de données. Ils sont accessibles via l'API GraphQL;
2. Un gestionnaire de corpus a été créé, il permet de créer, de modifier ou de supprimer un corpus;
3. Les articles d'un corpus peuvent être ordonnés depuis le gestionnaire de corpus; 
4. Les articles peuvent être associés à un corpus depuis le gestionnaire d'articles;

Ce fonctionnement a été implémenté dans Stylo lors du déploiement de la [version 2.1.0](https://github.com/EcrituresNumeriques/stylo/releases/tag/v2.1.0) de Stylo. 

La fonctionnalité _Corpus_ est maintenant dotée d'une structure grâce à laquelle nous pouvons implémenter une variété de comportements et de scénarios d'exports pour créer divers artefacts.

## Problématisation

(version courte)

Qu'est-ce qu'un ensemble de documents dans Stylo?

---

(version plus longue)

Jusqu'au stade du déploiement de Stylo 2.1.0, l'éditeur de texte n'était pas adapté pour créer des objets à une autre échelle que l'unité éditoriale (un article, un chapitre, etc.).
Stylo était d'ailleurs orienté pour écrire des articles de revue plus que des chapitres de livre ou de thèse, en témoigne l'ensemble des métadonnées disponibles dans le volet de métadonnées (directeur.ice.s de numéro, nom de la revue, etc.)

La création d'artefacts à une autre échelle éditoriale usait jusque-là de _tweaks_ pour tordre certaines fonctionnalités déjà existantes.
La fonctionnalité _Book_ décrite précédemment permettait de produire des livres tandis que pour produire un numéro de revue, il suffisait de remplir manuellement toutes les métadonnées et d'exporter un à un les articles pour reconstituer le numéro dans un autre espace que celui de Stylo.
Ainsi, pour les numéros de revue, il s'agissait de recompiler les sources (Markdown, YAML, BibTeX) ou leur pendant XML (par exemple) pour produire le nméro de revue en entier.

C'est d'ailleurs un comportement que nous souhaitons préserver dans Stylo : Stylo est pensé pour des articles individuels. L'unité est l'article. Cela implique que par exemple des informations sur les numéros se trouvent dans les métadonnées de chaque article.
Ce comportement est toujours souhaitable car un.e utilisateur.ice pourrait ne vouloir créer qu'un seul article (avant soumission à une revue) et remplir la totalité des métadonnées nécessaires à la publication sans devoir pour autant passer par la fonctionnalité _Corpus_ pour accéder à ces métadonnées.

Cette structuration des informations occasionne une perturbation de ce que peut être un document dans Stylo puisque nous retrouvons à l'intérieur de l'espace dédié à l'unité éditoriale des métadonnées destinées à décrire d'autre jeux d'échelle (d'abord le numéro de revue, puis le niveau au-dessus, la revue, voire même un autre niveau encore avec le diffuseur).

En ce sens nous pouvons d'abord nous demander ce qu'est un document dans Stylo.

### Qu'est-ce qu'un document ?

La notion de document renvoie à tout une histoire de la _documentation_.
Héritière de la _bibliographie_, la documentation désignait lors de la première _documentarisation_ (Salaun, 2004 ; Pedauque, 2006) au tournant du XX^e^ siècle un ensemble d'informations -- communéments appelées _documents_ -- accompagné d'un système d'organisation et de recherche des connaissances (Otlet, 2015 ; Briet, 1951).

Le document est un objet délicat à définir tant il revêt des formes différentes.
C’est certainement lié au fait que l’emploi du terme _document_ désigne une multitude d’objets que nous manipulons au quotidien dans différents lieux et espaces sociaux, que ce soit sur le lieu de travail, à l’école ou à l’université, dans des administrations publiques ou encore chez soi.
Un document peut très bien prendre l’apparence d’un formulaire administratif laissé sur le coin d’un bureau ou un livre bien ordonné sur une étagère d’une bibliothèque ou encore une photographie stockée dans les locaux des Archives nationales.
La notion de document est finalement un terme très générique que l’on a du mal à circonscrire du fait qu’il puisse désigner autant d’objets divers.

D’un point de vue étymologique, le dictionnaire latin-français _Gaffiot_ nous indique que _documentum_ signifie « exemple, modèle, leçon, enseignement, démonstration », définition qui ne fait pas forcément référence à un support de l’information alors que dans son acception plus contemporaine l’on désigne un document par celui-ci.

Paul Otlet, une figure importante durant la première documentarisation, a quant à lui défini le document comme suit : 

> 1. Livre (Biblion ou Document ou Gramme) est le terme conventionnel employé ici pour exprimer toute espèce de documents. Il comprend non seulement le livre proprement dit, manuscrit ou imprimé, mais les revues, les journaux, les écrits et reproductions graphiques de toute espèce, dessins, gravures, cartes schémas, diagrammes, photographies, etc. La Documentation au sens large du terme comprend : Livre, éléments servant à indiquer ou reproduire une pensée envisagée sous n’importe quelle forme.
> 2. Le Livre ainsi entendu présente un double aspect : a) il est au premier chef une oeuvre de l’homme, le résultat de son travail intellectuel ; b) mais, multiplié à de nombreux exemplaires, il se présente aussi comme l’un des multiples objets créés par la civilisation et susceptible d’agir sur elle ; c’est le propre de tout objet ayant caractère corporel et agencé techniquement.
> - (Otlet, 2015, p.9)

Si l'on continue notre lecture d'Otlet, une page plus loin, il nous donne les éléments consitutifs d'un document, à savoir : 

> Qu’est-ce qui dans le Livre lui est propre, qu’est-ce qui est proprement bibliographique ? On a déjà dit la distinction entre : a) la Réalité objective, b) la Pensée subjective ou l’état de conscience provoqué ou le moi par la réalité, c) la Pensée objective qui est l’effort de la réflexion combinée et collective sur ces données premières jusqu’à la science impersonnelle et totale, d) la Langue, instrument collectif de l’expression de la Pensée. Collection totale, tout livre contient ces quatre éléments associés concrètement en lui-même et que, par abstraction seulement, il est possible de dissocier et d’étudier à part. Ce qui est propre au livre, c’est le cinquième élément : la pensée désormais fixée par l’écriture des mots ou l’image des choses, signes visibles, fixés sur un support matériel.
> - (Otlet, 2015, p.10)

Nous pouvons résumer la pensée d'Otlet par cette dernière citation :

> Le définition la plus générale qu’on puisse donner du Livre et du Document est celle-ci : un support d’une certaine matière et dimension, éventuellement certain pliage ou enroulement sur lequel sont portés des signes représentatifs de certaines données intellectuelles.
> - (Otlet, 2015, p. 43)

Cette défintion du document fait apparaître des éléments que l'étymologie ne contient pas.
Il y aurait, d'abord, la pensée de l'auteur puis un support qui accueille cette pensée.

Donker Duyvis ajoute même qu'il faut prêter une attention toute particulière à ce que le support, loin d'être neutre dans cette affaire, ne déforme pas la pensée de l'auteur : 

> A document is the repository of an expressed thought. Consequently its contents have a spiritual character. The danger that blunt unification of the outer form exercises a repercussion on the contents in making the latter characterless and impersonal, is not illusory…. In standardizing the form and layout of documents it is necessary to restrict this activity to that which does not affect the spiritual contents and which serves to remove a really irrational variety
> - (Donker Duyvis, 1942. Translation from Voorhoeve, 1964, 48 dans Buckland, 1997)

Quelques décennies plus tard, en 1951, Suzanne Briet, dans son essai _Qu'est-ce que la documentation_, semble en décaccord avec Otlet, qu'elle qualifie de «mage», ainsi que son projet d'un Répertoire Bibliographique Universel de «chimère [...] qui n'offrait pas un intérêt comparable au plus localisé des catalogues collectifs».

Elle dresse une liste des définitions du document à partir de l'étymologie latine («enseignement ou preuve»), qu'elle complète en citant «une bibliographe contemporaine soucieuse de clarté» : «un document est une preuve à l'appui d'un fait».

Elle mentionne également une définition officielle donnée par l'Union Française des Organismes de Documentation : «toute base de connaissance fixée matériellement et susceptible d'être utilisée pour consultation, étude ou preuve».

En confrontant ces définitions au regard de la linguistique et de la philosophie, une définition plus abstraite émerge : «tout indice concret ou symbolique, conservé ou enregistré, aux fins de représenter, de reconstituer ou de prouver un phénomène physique ou intellectuel».

À partir de cette définition du document, elle utilise l'exemple, aujourd'hui très célèbre, de l'antilope.
Est-ce qu'une antilope est un document ? Oui et non, cela dépend du contexte.
Si l'antilope est dans son environnement naturel, il ne s'agit que d'une antilope.
Par contre, si l'antilope est capturée et mise en cage dans un zoo, elle est alors cataloguée et devient un document vivant.
Lorsqu'elle mourra, l'antilope sera empaillée pour conservation puis exposée dans un musée.
Elle sera alors un nouveau document.
Si de l'antilope nous prenons des photographies, rédigeons des articles de presse ou savant, réalisons des enregistrements sonores ou des vidéos, tous ces supports vont capter un fragment de l'antilope et seront ce que Briet nomme des documents secondaires.

À la différence d'Otlet, le document de Briet ne correspond pas à l'impression de la volonté d'un auteur sur un support.
Pour elle, le document est une preuve qu'on insère dans une architecture de l'information.
La preuve fait alors partie d'un catalogue d'où l'on peut extraire des informations depuis une unité de ce catalogue : le document.

Que ce soit chez Otlet ou chez Briet, les deux conceptions du document ont en commun de mobiliser une action humaine.
Dans le premier cas il s'agit de l'action réalisée par l'auteur d'un document tandis que dans le second il s'agit de la création d'une base de connaissances organisées.

### Qu'est-ce qu'un document numérique ?

En 1997, Buckland s'intéresse aux changements qu'apporte le numérique à la nature de ce document.
À la suite d'une revue de la littérature mentionnant les travaux d'Otlet et de Briet, il montre que certaines conceptions du documents sont très limitées, comme celle du théoricien Ranganathan :

> But they are not documents; because they are not records on materials fit for handling or preservation. Statues, pieces of china, and the material exhibits in a museum were mentioned because they convey thought expressed in some way. But none of these is a document, since it is not a record on a more or less flat surface." (Ranganathan, 1963)

Cette conception du document manipulable sur une surface plane est très limitée, et ne permet pas d'y inclure beaucoup d'autres supports que le papier.
Dès les années 1970, l'informatique perturbe cette notion du document avec sa représentation du monde (_worldview_) numérique.
Néanmoins, la bataille entre les fournisseurs de logiciels d'écriture nous montre que c'est bien le paradigme graphique qui l'emporte durant les années 1980, notamment grâce à l'apparition de la page à l'écran (Bergin, 2006a, 2006b ; Kirschenbaum, 2016) et des interfaces WYSIWYG.

Seulement, cette page affichée à l'écran n'est ni un objet pérenne, ni un objet manipulable.
Elle n'est qu'une projection de ce à quoi elle pourrait ressembler une fois imprimée (donc sortir de son écosystème numérique).
En 1994, David Lévy traite justement de la variabilité des écritures numériques.
Dans un article intitulé «Fixed or fluid?», il critique la distinction entre un document papier _fixe_ et un document numérique (qui n'en serait pas un) fluide ou variable.
Sa thèse consiste plutôt à considérer que tous les types de documents peuvent être à la fois fixe et variable. Il nous rappelle également qu'un document numérique existe matériellement sous forme de bits dans la support de la mémoire numérique.

Pour le dire autrement, Lévy nous dit finalement que ce n'est pas le support qui compte dans la détermination d'un document.
À ce sujet, Buckland remet le propos de Lévy en tension avec la discussion entre Otlet et Briet et nous rappelle que ce débat autour du support a déjà eu lieu (et a déjà été résolu).

En France, au début des années 2000, un collectif de chercheur.e.s se forme sous le nom de Roger T. Pédauque et travaille pendant plusieurs années à la définition du document numérique.
Pour Pédauque, si le début du XX^e^ marque la première documentarisation, la globalisation de l'informatique et à la mise en réseau des informations (Internet avec des paplications comme les courriels, le Web, etc.) marquent quant à elle la deuxième documentarisation.
Ce collectif défini le document (numérique) par une équation relativement simple (Pedauque, 2006) : 

$$document = support + inscription$$

Selon Pédauque, le document est composée de 3 dimensions , selon d'où proviennent les chercheurs du collectif: 

> - Le document comme forme ; sous cette rubrique, nous rangerons les approches qui analysent le document comme un objet, matériel ou immatériel et qui en étudient la structure pour mieux l'analyser, l'utiliser ou le manipuler.
> - Le document comme signe ; pour ces chercheurs le document est perçu avant tout comme porteur de sens et doté d'une intentionnalité ; ainsi le document est indissociable du sujet en contexte qui le construit ou le reconstruit et lui donne sens ; en même temps, il est pris dans un système documentaire ou un système de connaissances.
> - Le document comme médium ; cette dimension enfin pose la question du statut du document dans les relations sociales ; le document est une trace, construite ou retrouvée, d'une communication qui s'est affranchie de l'espace et du temps ; en même temps, il est un élément de systèmes identitaires et un vecteur de pouvoir.

Cette approches du document sont reformulées par Pédauque au prisme des différentes technologies émergentes dans les années 2000 (Web sémantique notamment).

Ainsi, Pédauque remet sur le support sur la table de la discussion et réamorce lui aussi le vieux débat entre Otlet et Briet.



Un mouvement similaire à Pédauque émerge en 2001 de l'autre côté de l'Atlantique.
À l'initiative de Maribeth Back et Niels Windfeld Lund, c'est la _Document Academy_ qui voit le jour et s'inscrit dans les _document studies_.
Contrairement à Pédauque qui est arrivé en bout de financement à la fin des années 2000, la Document Academy perdure encore jusqu'à 2024. (Des liens ont d'ailleurs été tissés entre les collectifs.)^[On peut trouver leurs travaux à cette URL : <https://ideaexchange.uakron.edu/docam/>)

Tous ces travaux élargissentla notion de document pour qu'elle englobe le numérique et précisent également les spécificités du document numérique (en tant que type de document) en comprenant certaines propriétés comme l'hypertextualité, etc.

### Les modèles textuels

L'une des particularités du document numérique est la capacité à représenter et à structurer l'information à l'intérieur d'un espace délimité à partir d'une modélisation textuelle et explicite, que l'on peut déclarer sous forme de balises (_markup_).

Historiquement, les premiers outils d'écriture numérique ([_Electric Pencil_](http://www.trs-80.org/electric-pencil/), [_Roff_](https://www.man7.org/linux/man-pages/man7/roff.7.html)) n'avait pas d'interface graphique.
Dès lors la mise en page d'un document dans un terminal se faisait au moyen d'un balisage du texte pour déclarer explicitement les chaînes de caractères auxquelles le logiciel devait appliquer les commandes de mise en page.
Donald Knuth développe TeX sur le même principe : chaque ligne du document correspond à une commande que le logiciel interprète pour mettre en page les informations.

Ce principe de déclaration des balises ne sert pas uniquement à déclarer des commandes dans le texte.
En parallèle de ce fonctionnement pour l'impression, un employé d'IBM, Charles Goldfarb, commence à faire développer des 1969 un langage de description des éléments dans le document, il s'agit de GML (_Generalized Markup Language_).

Des applications comme SGML (_Standard Generalized Markup Language_) en 1986 puis HTML en 1989 (_HyperText Markup Language_) voient le jour et s'appuient sur la modélisation théorique du texte dans un document OHCO (_ordered hierarchy of content object_)(DeRose, 1990).
Les premiers langages de description des informations se répandent dans toute la sphère numérique, notamment avec le Web.
Si HTML impose des règles de grammaire décidé par la communauté et le W3C (bien que ce modèle soit très permissif), ce n'est pas le cas de SGML qui permet à tout un chacun de créer son propre modèle textuel.
Dix ans plus tard (1999), les spécifications d'un nouveau langage de balisage sont publiées, il s'agit d'XML (_Extensible Markup Language_).
Héritier direct de SGML, XML garde cette souplesse de la modélisation des informations et s'enrichit d'un grand nombre de schémas différents ou de communautés différentes (on pense à toute la communauté TEI (_Text Encoding Initiative_).

Au début des années 2000, ces formats apparaissent trop verbeux.
On voit l'émergence de "langages de balisage léger" comme la YAML ou le Markdown.
Si le Markdown est une version allégée de HTML, le YAML est un format utilisée pour structurer des données du type métadonnées ou informations de configuration.

Si le XML permet de faire ces deux choses à la fois, on remarque qu'il y a aussi des pratiques où l'on détache les métadonnées d'un contenu plus textuel (un fichier markdown peut être associé à un fichier de métadonnées yaml).
D'autres formats voient le jour (exemple JSON) pour répondre à d'autres besoin de structuration des données.

Tous ces formats nous montrent que le document n'est plus seulement "un seul objet" ou tous les feuillets sont assemblé et relié physiquement entre eux.
Le numérique, grâce à son hypertuextualité, permet de dissocier et d'organiser les informations entre plusieurs fichiers.
Un document n'est plus seulement un objet du type livre, revue, album de photographies, collection de cassettes, mais dispose les informations en divers endroits afin de répondre à des besoins différents.

La montée en volume des données durant les années 2000 et l'arrivée du Big Data et des bases de données, qu'elles soient relationnelles ou non, fragmente encore plus les informations.

À ce stade, la question du support n'a plus aucun sens, nous dit Bachmimont (2022).
Il nous propose cependant de refonder l'herméneutique documentaire en s'appuyant sur 4 niveaux : 

> Au premier niveau, celui du code, le numérique s’interprète de manière formelle en algorithme, modèles mathématiques et logiques, où la manipulation binaire aveugle se comprend comme un calcul et une preuve opérant sur une logique symbolique. C’est le niveau des théoriciens du calcul, de l’informatique théorique, et de l’algorithmique.
> Au second niveau, celui des formats, le code s’interprète selon des conventions permettant de le mettre en correspondance avec des domaines du réel : le binaire correspondra à des nombres, à des caractères, à des pixels, à des points d’impression, etc. A ce niveau, technique et logique, le code n’est plus seulement une réalité mathématique et formelle permettant d’interpréter ce qui se passe dans les registres binaires, mais devient une manipulation du réel où l’on voit des écrans afficher des contenus, des textes se manipuler, des sons se produire.
> Au troisième niveau, celui des applications, les formats se mobilisent dans des systèmes logiciels s’intégrant dans des logiques métiers, des usages, des pratiques. Un système est conçu pour permettre un usage.
> A ces trois niveaux, il convient d’en ajouter un, ou de le penser de manière sous-jacente, c’est celui de la concrétisation matérielle et physique de la machine informatique, qui définira une architecture, un code machine natif (un microprocesseur), qui sera la base sous-tendant toutes les autres opérations

Le document n'est alors plus seulement constitué d'un format d'encodage et d'informations mais de toute le système qui en permet l'existence, allant de l'architecture du micro-processeur, aux systèmes d'indexation et de stockage des informations et des données en passant par les algorithmes qui permettent de les trier, les récupérer, les traiter, et les assembler dans une forme lisible pour les individus. 

### Définir le document dans Stylo

Au prisme de cette définition du document (numérique), nous pouvons définir le document dans Stylo par une incarnation matérielle formée de : 

1. un composite de 3 sources délimitées par des formats (Markdown, YAML, BibTeX)
2. une organisation des informations en base de données (MongDB)
3. un système de recherche incarné dans des requêtes GraphQL (pour réordonner les informations et les soumettre à l'utilisateur.ice)

Ce sont ces médiations dans l'environnement qu'est Stylo qui déterminent le document dans Stylo.
En somme, les différences de jeux d'échelle éditoriale n'ont à ce stade aucun d'impact dans la façon de définir un document dans Stylo et cela ne posait pas de problème jusqu'à l'apparition d'une nouvelle entité dans cet espace, le _Corpus_.

Un document comme ce billet de discussion est donc délimité dans Stylo par le texte en Markdown, les métadonnées en YAML : 

```yaml
---
articleslies:
  - auteur: ''
    title: ''
    url: ''
date: '2024-11-01'
director:
  - foaf: ''
    forname: ''
    gender: ''
    isni: ''
    orcid: ''
    surname: ''
    viaf: ''
dossier:
  - id: ''
    title_f: ''
lang: fr
link-citations: true
nocite: '@*'
rights: Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
title_f: réflexion corpus
translations:
  - lang: ''
    title: ''
    url: ''
translator:
  - forname: ''
    surname: ''
---
```

Des éléments tels que `director` ou `translator` sont explicitement déclarés dans cette partie du document.
On y identifie un `director` avec sept informations différentes tandis qu'un `translator` n'en comporte que deux.
La structure des informations contraint ici les auteurs à ne saisir que les informations que l'application peut traiter.
Ces deux entités sont déterminées par cette structure des informations (si l'auteur les remplie).

Autre élément que nous trouvons dans ce document : une requête `POST` en JSON comportant des informations pour mettre à jour la base de données : 

```JSON
query
: 
"query updateWorkingVersion($articleId: ID!, $content: WorkingVersionInput!) {\n  article(article: $articleId) {\n    updateWorkingVersion(content: $content) {\n      updatedAt\n    }\n  }\n}"
variables : {userId: "61d62c4978651b001208b7aa", articleId: "6723a6ab33b8f40011f9c8b6", content: {,…}}
articleId : "6723a6ab33b8f40011f9c8b6"
content : {,…}
    md : "## Introduction : du _Book_ au _Corpus_ \n\nLa fonctionnalité [...]
userId : "61d62c4978651b001208b7aa"
```
Les informations sont restructurées par les algorithmes de Stylo et font apparaître de nouvelles notions comme `WorkingVersion` qui lui sont constitutives.

La `WorkingVersion`, ou version de travail, correspond à l'espace en cours d'édition dans l'interface de Stylo.
Cette requête permet entre autre de sauvegarder l'écriture en cours et limite la perte de données en cas de coupure du réseau (On ne l'a pas dit mais Stylo est un environnement Web, si on coupe internet, couic plus de stylo).

Pourquoi l'appeler _version_ et non pas _copie_ ?

S'il y a une version de travail, c'est pour la distinguer d'autres version que l'utilisateur peut déclarer manuellement via le gestionnaire de version.
Ce système de versionnement des informations est inspiré des protocoles de versionnement utilisés principalement dans l'univers du développement (donc pour des fichiers contenant du code).

On pense au protocole distribué comme `git`, ou encore un protocole centralisé comme `subversion`, etc.

Dans Stylo, un même document embarque alors plusieurs versions de lui-même (avec ces sources YAML, Markdown, BibTeX) et tout l'historique des modifications que l'on y apporte.
De ce principe découle une fonctionnalité qui permet de générer les différences entre les versions (une forme de représentation de l'historique des modifications).

Maintenant que nous avons une vue d'ensemble sur ce qu'est un document Stylo, nous nous demandons ce que peut être un ensemble de documents Stylo.

Pour le dire autrement, il ne s'agit pas de faire une collection de multiples documents mais de créer un méta-document à partir de plusieurs documents (que l'on considère alors comme des unités documentaires).

Par exemple, le livre est une forme méta-documentaire comportant plusieurs documents : des chapitres.
Hors il ne s'agit là que d'une forme possible, et dans Stylo nous cherchons à représenter ce nouveau document sous la forme générique de _Corpus_ applicable à un ensemble plus vaste de documents. 


### Qu'est-ce qu'un ensemble de documents dans Stylo ?

Nous avons deux cas d'usages principaux avec des sous-cas:

- [Corpus d'articles](https://github.com/EcrituresNumeriques/stylo/pull/1058)
  1. pour un dossier. Exporter un dossier de revue, par exemple en XML Métopes, pour publication en lot
  2. pour une revue entière: tous les articles d'une revue, par exemple pour produire automatiquement le site d'une revue
- Corpus de chapitres
  1. pour faire un livre
  2. pour faire un mémoire ou une thèse

## Bibliographie

Bachimont, B. (2022). Donner du sens aux données : Les ruses du numérique : Les disciplines du document face à la métis du calcul. Interfaces numériques, 11(2), Article 2. https://doi.org/10.25965/interfaces-numeriques.4838
Bergin, T. J. (2006a). The Origins of Word Processing Software for Personal Computers : 1976-1985. IEEE Annals of the History of Computing, 28(4), 32‑47. IEEE Annals of the History of Computing. https://doi.org/10.1109/MAHC.2006.76
Bergin, T. J. (2006b). The Proliferation and Consolidation of Word Processing Software : 1985-1995. IEEE Annals of the History of Computing, 28(4), 48‑63. IEEE Annals of the History of Computing. https://doi.org/10.1109/MAHC.2006.77
Briet, S. (1951). Qu’est-ce que la documentation ? (Editions Documentaires Industrielles et Techniques). http://martinetl.free.fr/suzannebriet/questcequeladocumentation/
Buckland, M. K. (1997). What is a “document”? Journal of the Association for Information Science and Technology, 48(9), 804‑809. [https://doi.org/10.1002/(sici)1097-4571(199709)48:9<804::aid-asi5>3.0.co;2-v](https://doi.org/10.1002/(sici)1097-4571(199709)48:9%3C804::aid-asi5%3E3.0.co;2-v)
DeRose, S. J., Durand, D. G., Mylonas, E., & Renear, A. H. (1997). What is text, really? ACM SIGDOC Asterisk Journal of Computer Documentation, 21(3), 1‑24. https://doi.org/10.1145/264842.264843
Kirschenbaum, M. G. (2016). Track Changes : A Literary History of Word Processing (Illustrated edition). Belknap Press: An Imprint of Harvard University Press.
Levy, D. M. (1994). Fixed or fluid? Proceedings of the 1994 ACM European Conference on Hypermedia Technology - ECHT &#39;94. https://www.academia.edu/103818913/Fixed_or_fluid
Lund, N. W., & Buckland, M. (2008). Document, documentation, and the Document Academy : Introduction. Archival Science, 8(3), 161‑164. https://doi.org/10.1007/s10502-009-9076-3
Otlet, P. (2015). Le livre sur le livre : Traité de documentation. Les Impressions nouvelles.
Pédauque, R. T. (2006). Le document à la lumière du numérique. C&F éditions.
Pédauque, R. T. (2007). La redocumentarisation du monde. Editions Cépaduès.
Salaün, J.-M. (2004). Chronique inachevée d’une réflexion collective sur le document. Communication et langages, 140(1), 9‑17. https://doi.org/10.3406/colan.2004.3263
